build.gradle

```diff
@@ -562,19 +563,20 @@ task testDocker(type: Exec) {
 
 task dockerUpload(type: Exec) {
   dependsOn distDocker
+  def dockerBuildVersion = project.hasProperty('release.releaseVersion') ? project.property('release.releaseVersion') : "${rootProject.version}"
   def imageName = "hyperledger/besu"
   def azureImageName = "hyperledger.azurecr.io/besu"
-  def image = project.hasProperty('release.releaseVersion') ? "${imageName}:" + project.property('release.releaseVersion') : "${imageName}:${project.version}"
+  def image = "${imageName}:${dockerBuildVersion}"
   def cmd = "docker push '${image}'"
   def additionalTags = []
 
   if (project.hasProperty('branch') && project.property('branch') == 'master') {
     additionalTags.add('develop')
   }
 
-  if (! version ==~ /.*-SNAPSHOT/) {
+  if (! dockerBuildVersion ==~ /.*-SNAPSHOT/) {
```

Comment:
```suggestion
  if (dockerBuildVersion !=~ /.*-SNAPSHOT/) {
```

Reply:
Syntax ness of this is interesting:
`~ is a pattern, =~ is a matcher and ==~ gives a boolean` but you cant directly negate a match so I've put the whole thing in brackets